### PR TITLE
[batch] Mitigation for case insensitive billing projects and usernames

### PR DESCRIPTION
**Pull request by jigold**
FYI @danking

### CODE REVIEW

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
This looks good, just one thing that felt odd. Additionally, I just want to make sure that we're planning not to add any new users/billing projects in the time between migration and deploy for this PR right? I don't think a fully-safe multi-step PR is worth it, just checking to see if that was the plan

**Comment by daniel-goldstein** [comment]
One more thing that just popped into mind: should there be indexes on these new columns? Probably doesn't matter much on the billing_projects table but I was just wondering

**Review by daniel-goldstein — APPROVED** [review]

File: batch/sql/case-sensitive-billing-project.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+UPDATE billing_project_users
+SET user_cs = user;
+
+ALTER TABLE billing_projects MODIFY COLUMN name_cs VARCHAR(100) NOT NULL COLLATE utf8mb4_0900_as_cs;
+ALTER TABLE billing_project_users MODIFY COLUMN user_cs VARCHAR(100) NOT NULL COLLATE utf8mb4_0900_as_cs;
+
+CREATE UNIQUE INDEX `billing_project_name_cs` ON `billing_projects` (`name_cs`);
+CREATE INDEX `billing_project_name_cs_status` ON `billing_projects` (`name_cs`, `status`);
+CREATE INDEX `billing_project_users_billing_project_user_cs` ON `billing_project_users` (`billing_project`, `user_cs`);
```
Comment (danking) [inline batch/sql/case-sensitive-billing-project.sql:15]: What's the plan for dropping the old columns and fixing up the foreign key references?

**Review by danking — DISMISSED** [review]

File: batch/sql/case-sensitive-billing-project.sql (hunk 1 above)
Comment (jigold) [reply batch/sql/case-sensitive-billing-project.sql:15]: There is no plan to do so. I wrote out what needs to happen and it's complicated, is going to take a lot of work / testing to get right, and didn't seem as important as other things we could be doing. If you feel differently, then I can go ahead and do it.

Comment (jigold) [reply batch/sql/case-sensitive-billing-project.sql:15]: I forgot how far I got on this, but here was a pilot branch I believe: https://github.com/hail-is/hail/commit/32fe47906dc7c189703ba713cf0e3794084af475

**Comment by jigold** [comment]
@danking bump

**Comment by jigold** [comment]
@danking Can we come up with a plan for whether this short term fix is ok or whether you want me to do the full transformation?

**Comment by danking** [comment]
Ah sorry this wasn't in my CI queue

**Comment by danking** [comment]
We should put the mitigation in and also start a separate PR/project to do the proper fix. Batch is pretty complex and hard to manage already, let's make sure we don't leave cruft in for long periods. I think having this migrated properly in the next 2-3 months is reasonable.

**Review by danking — APPROVED** [review]

**Pull request merged**